src/Bridge/Doctrine/Orm/CollectionDataProvider.php

```diff
@@ -52,7 +53,7 @@ public function getCollection(string $resourceClass, string $operationName = nul
         $queryBuilder = $repository->createQueryBuilder('o');
 
         foreach ($this->collectionExtensions as $extension) {
-            $extension->applyToCollection($queryBuilder, $resourceClass, $operationName);
+            $extension->applyToCollection($queryBuilder, new QueryNameGenerator(), $resourceClass, $operationName);
```

Comment:
The `QueryNameGenerator` should be instantiated only once before the loop and shared across all extensions.